### ISSUE TITLE
Allow `file_format` to be lower-case

### ISSUE THREAD

**Issue by Fokko** [apache/iceberg-python#1340]
### Feature Request / Improvement

Based on some testing in Iceberg-Rust https://github.com/apache/iceberg-rust/pull/349#pullrequestreview-2444443544 it came to the light that with PyIceberg the `file_format` field has to be uppercase. I think it would be good to allow for lowercase as well 👍

**Comment by kevinjqliu** [comment]
https://github.com/apache/iceberg-python/blob/93ebd39e3c457dcb86cd053c60d2d13f0713a637/pyiceberg/manifest.py#L95-L102

**Comment by hgollakota** [comment]
Hey, gonna submit a pull-request - here's the solution I'm proposing:

```
class FileFormat(str, Enum):
    AVRO = "AVRO", "avro"
    PARQUET = "PARQUET", "parquet"
    ORC = "ORC", "orc"

    def __new__(cls, value, *value_aliases):
        obj = str.__new__(cls)
        obj._value_ = value
        for alias in value_aliases:
            cls._value2member_map_[alias] = obj
        return obj

    def __repr__(self) -> str:
        """Return the string representation of the FileFormat class."""
        return f"FileFormat.{self.name}"
```

**Comment by jiakai-li** [comment]
Not sure if I'm still able to work on this or not. But I submitted another PR, rather than modifying `__new__`, I think `_missing_` is a better candidate given its semantic meaning.

**Comment by Fokko** [comment]
Fixed in https://github.com/apache/iceberg-python/pull/1411 🙌

**Comment by kevinjqliu** [comment]
Following up on this, #1362 was merged and then reverted, #1518. Ultimately, #1411 resolves this issue, and there's a test to ensure the behavior 
https://github.com/apache/iceberg-python/blob/985029042199d870f25b6fbec0e80907d4440f41/tests/utils/test_manifest.py#L609-L629
